cmake/toolchains/riscv64.cmake

```diff
@@ -0,0 +1,21 @@
+
+# Prevent double invocation.
+if (RISCV_TOOLCHAIN_INCLUDED)
+    return()
+endif()
+set(RISCV_TOOLCHAIN_INCLUDED TRUE)
+
+set(CMAKE_SYSTEM_NAME Linux)
+set(CMAKE_SYSTEM_VERSION 1)
+set(CMAKE_SYSTEM_PROCESSOR riscv)
+
+# User can use -DCMAKE_FIND_ROOT_PATH to specific toolchain path
+set(CMAKE_C_COMPILER ${CMAKE_FIND_ROOT_PATH}/bin/riscv64-unknown-linux-gnu-clang)
+set(CMAKE_CXX_COMPILER ${CMAKE_FIND_ROOT_PATH}/bin/riscv64-unknown-linux-gnu-clang++)
+set(CMAKE_LINKER ${CMAKE_FIND_ROOT_PATH}/bin/riscv64-unknown-linux-gnu-ld)
```

Comment:
Why there is `unknown` in the name of toolchain programs. Does official toolchain come like that?

Reply:
This `riscv64-unknown-linux-gnu-` is the convention prefix provided by [riscv-gnu-toolchain](https://github.com/riscv-collab/riscv-gnu-toolchain)(This repo is in riscv org). I can make the prefix be a cache variable if you think it's better.

Reply:
If we want to support `riscv32`, I think it's better to create another toolchain.cmake like `riscv32.cmake`. It will avoid adding additional variable to select 32bit/64bit target.